src/documents/views/VersionPreviewView.js

```diff
@@ -0,0 +1,285 @@
+import React, {useState, useEffect} from 'react';
+import {Helmet} from 'react-helmet';
+import {useQuery} from '@apollo/react-hooks';
+import {GET_FILE_BY_ID} from '../queries';
+import {fileSortedVersions} from '../utilities';
+import {
+  Button,
+  Container,
+  Header,
+  Message,
+  Segment,
+  Popup,
+  Grid,
+  Icon,
+  Label,
+} from 'semantic-ui-react';
+import TimeAgo from 'react-timeago';
+import {longDate} from '../../common/dateUtils';
+import Papa from 'papaparse';
+import * as XLSX from 'xlsx';
+import {LogOnMount} from '@amplitude/react-amplitude';
+import {HotTable} from '@handsontable/react';
+import 'handsontable/dist/handsontable.full.css';
+
+const VersionPreviewView = ({match}) => {
+  const [fullWidth, setFullWidth] = useState(false);
+  const [displayLength, setDisplayLength] = useState(0);
+  const [loading, setLoading] = useState(false);
+  const [data, setData] = useState([]);
+  const [errors, setErrors] = useState([]);
+  const {data: fileData} = useQuery(GET_FILE_BY_ID, {
+    variables: {kfId: match.params.fileId},
+  });
+  const file = fileData && fileData.fileByKfId;
+  const fileName = file && file.name;
+  const version =
+    file &&
+    file.versions.edges.find(v => v.node.kfId === match.params.versionId).node;
+  const latestVersionId = file && fileSortedVersions(file)[0].node.kfId;
+  const downloadURL = version && version.downloadUrl;
+
+  const getData = async limiteLength => {
+    setLoading(true);
+
+    if (version.fileName.endsWith('csv') || version.fileName.endsWith('tsv')) {
+      const resp = await fetch(downloadURL, {
```

Comment:
You seem to repeat this twice for csv,tsv and xlsx,xls. Maybe just do it once and handle the differences elsewhere?